### PR TITLE
fix(docs): Remove outdated network endpoints and add endpoint check action

### PR DESCRIPTION
**Pull request by MalteHerrmann**
This PR adds a script to check the integrity of a selection of listed publicly available endpoints.
Based on the checked results, I have removed a bunch of outdated entries from the mainnet and testnet lists.

## Release Notes

- **New Features**
	- Introduced a new GitHub Actions workflow for automated API endpoint checks.
	- Added structured documentation for publicly available endpoints related to the Evmos blockchain.

- **Bug Fixes**
	- Updated and consolidated mainnet and testnet endpoint addresses, removing outdated entries.

- **Documentation**
	- Updated the `requirements.txt` file to include new dependencies for improved functionality.

- **Tests**
	- Added unit tests for endpoint extraction and query functionalities to ensure reliability.

### CODE REVIEW

File: scripts/check_endpoints/endpoints.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        if "### Mainnet" in line:
+            extract_mainnet = True
+            extract_testnet = False
+            continue
+        elif "### Testnet" in line:
+            extract_mainnet = False
+            extract_testnet = True
+            continue
+        elif line[0] != "|":
+            continue
```
Comment (0xstepit) [inline scripts/check_endpoints/endpoints.py:62]: Isn't this already covered by line 50?

File: scripts/check_endpoints/queries.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+call a simple query on a given endpoint
+based on the type of endpoint.
+"""
+
+from endpoints import Endpoint
+import requests
+from typing import Union
+
+
+def query_endpoint(ep: Endpoint) -> Union[bool, None]:
```
Comment (0xstepit) [inline scripts/check_endpoints/queries.py:12]: What about websocket and Cosmos gRPC.

nit: not part of this PR, but it would be nice to rename Tendermint to Comet

**Review by 0xstepit — APPROVED** [review]
This is great! Nice job @MalteHerrmann, just added a couple of quesitons.

File: scripts/check_endpoints/queries.py (hunk 2 above)
Comment (MalteHerrmann) [reply scripts/check_endpoints/queries.py:12]: will create a follow-up issue to address the CometBFT names.
